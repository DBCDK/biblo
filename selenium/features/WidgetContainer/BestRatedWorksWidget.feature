Feature: Best Rated Works widget
  This widget displays a gallery of the best bibliographic works according to the review base.

  Scenario: The widget can load with a minimum of configuration
    # The previewPage requires that you're already on a page, so start by going to the frontpage.
    When mock aboutpage is loaded
    Given a user visits the aboutpage
    Then page is not error page

    # Load the relevant mock and open the preview page with a single widget configured
    When mock bestRatedWorksWidget is loaded
    Given PreviewPage is loaded with widgetConfig: [{"widgetName": "BestRatedWorksWidget", "widgetConfig": {"title": "this is a super cool way to show bibliographic works!", "size": 2, "showTitle": true}}]
      And the cookiewarning has been closed

    # Make some assertions based on what we expect.
    Then page is not error page
     And the page contains: THIS IS A SUPER COOL WAY TO SHOW BIBLIOGRAPHIC WORKS!
     And the page contains: Alle elsker Sigge
