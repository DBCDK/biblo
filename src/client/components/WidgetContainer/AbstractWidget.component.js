/**
 * @file:
 */

import {Component, PropTypes} from 'react';

/**
 * @Class: AbstractWidget
 * This class defines a few helper methods on top of React.Component to aid widgets.
 */
export class AbstractWidget extends Component {
  constructor(props) {
    super(props);

    // No props passed to super. Throw error to notify.
    if (!props) {
      throw new Error('No props declared! Did you remember to call super(props); ?');
    }

    // Could not find widgetActions. This should only happen if the component is initialized outside widget context.
    if (!props.widgetActions) {
      throw new Error('All widgets must have widgetActions!');
    }

    // Could not find method callServiceProvider!
    // This means we are not dealing with widgetAction, most likely an incomplete mock.
    if (!props.widgetActions.callServiceProvider) {
      throw new Error('Could not find callServiceProvider in widgetActions, invalid widget initialization!');
    }
  }

  /**
   * This function dispatches an event that gets caught by serviceProviderReduxMiddleware.
   * The middleware then sets up a listener and calls the biblo service provider with the transform event (passed in through the parameter).
   * The listener dispatches a redux action with type = `${event}Response`.
   * This redux action has data = resultFromServiceProvider.
   * @param {String} event - The event of the transform you want to call.
   * @param {*} query - The query sent to the transform.
   */
  callServiceProvider(event, query) {
    this.props.widgetActions.callServiceProvider(event, query);
  }

  render() {
    // This is an abstract component, not for rendering!
    throw new Error('Child class missing render function or calling super from render!');
  }
}

AbstractWidget.displayName = 'AbstractWidget';
AbstractWidget.propTypes = {
  // This is an object containing all the actions in widget.actions.js
  widgetActions: PropTypes.object.isRequired,

  // This is an object containing data as passed from the CMS.
  widgetConfig: PropTypes.object.isRequired,

  // This is a string, with the name of the widget location (for example: FrontPageTop)
  // This is how we specify where a specific widget goes.
  widgetLocationName: PropTypes.string.isRequired,

  // This is an object containing the property from the widgetReducer that matches the widgetName.
  widgetReducerProp: PropTypes.any,

  // This is the entire state from the widgetReducer.
  widgetState: PropTypes.object.isRequired
};
