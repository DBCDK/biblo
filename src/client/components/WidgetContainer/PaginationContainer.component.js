/**
 * @file: This file implements the pagination container, which abstracts pagination away
 */

import React, {Component, PropTypes} from 'react';

import VisibilitySensor from 'react-visibility-sensor';
import Icon from '../General/Icon/Icon.component';

import plusSvg from '../General/Icon/svg/functions/plus.svg';
import crossSvg from '../General/Icon/svg/functions/close.svg';
import loadingSvg from '../General/Icon/svg/spinners/loading-spin.svg';

import './scss/PaginationContainer.component.scss';

export class PaginationContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pageNumber: this.props.pageIncrements,
      loading: true
    };

    this.getShowMoreButton = this.getShowMoreButton.bind(this);
    this.getCloseButton = this.getCloseButton.bind(this);
    this.onGetNextPage = this.onGetNextPage.bind(this);
    this.onClosePagination = this.onClosePagination.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({loading: false});
  }

  onGetNextPage() {
    const nextPage = this.state.pageNumber + this.props.pageIncrements;
    this.props.nextPageFunction(nextPage);
    this.setState({
      pageNumber: nextPage,
      loading: true
    });
  }

  onClosePagination() {
    this.setState({
      pageNumber: this.props.pageIncrements
    });
  }

  getShowMoreButton() {
    if (this.props.lastPageIndex && this.props.lastPageIndex <= this.state.pageNumber) {
      return (
        <span className="show-more-button--no-more-pages" />
      );
    }

    return (
      <span
        className="show-more--pagination--button"
        onClick={this.onGetNextPage}
      ><Icon glyph={plusSvg} /> Vis Flere</span>
    );
  }

  getCloseButton() {
    return (
      <span
        className="close--pagination--button"
        onClick={this.onClosePagination}
      ><Icon glyph={crossSvg} /> Luk</span>
    );
  }

  render() {
    const showMoreButton = this.getShowMoreButton();
    const closeButton = this.getCloseButton();

    let width = 700;
    if (typeof window !== 'undefined') {
      width = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
    }

    let loading = <span />;
    if (this.state.loading && this.props.genericLoading) {
      loading = <Icon glyph={loadingSvg} height={50} width={50} />;
    }

    return (
      <div className="pagination-container">
        <div className="pages--container">
          {this.props.pages.slice(0, this.state.pageNumber)}
          {width < 600 ?
            <VisibilitySensor onChange={(vis) => (vis && this.onGetNextPage())} delayedCall={true}>
              <span className="small-screen-sensor"> &nbsp; </span>
            </VisibilitySensor>:
            <span className="large-screen" />
          }
        </div>

        <div className="pagination-controls--spinner">
          {loading}
        </div>

        <div className="pagination-controls--container">
          <hr className="pagination-controls--spacer" />
          {showMoreButton}
          {closeButton}
          <hr className="pagination-controls--spacer" />
        </div>
      </div>
    );
  }
}

PaginationContainer.displayName = 'PaginationContainer';
PaginationContainer.propTypes = {
  nextPageFunction: PropTypes.func.isRequired,
  pages: PropTypes.array.isRequired,
  pageIncrements: PropTypes.number,
  lastPageIndex: PropTypes.number,
  genericLoading: PropTypes.bool
};
PaginationContainer.defaultProps = {
  pageIncrements: 1,
  lastPageIndex: 0,
  genericLoading: false
};
