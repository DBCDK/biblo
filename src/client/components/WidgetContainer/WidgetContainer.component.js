/**
 * @file: WidgetContainer.component.js is the generic widget component which connect widgets to their data,
 * and renders them where they need to be rendered.
 *
 * @usage:
 * <WidgetContainer
 *   widgetLocationName="A string which describes the location"
 *   widgetState={[Object containing the output from the widgetReducer]}
 *   widgetActions{[Object containing all the methods related to widgets]}
 * />
 *
 */

import React, {Component, PropTypes} from 'react';
import widgetComponents from './widgets';

class WidgetContainer extends Component {
  getTitle(widgetConfig = {}, classNames = '') {
    let title = <span className="widget--no--title"/>;

    const titleClasses = `widget--title--generic ${classNames}`;
    if (widgetConfig.showTitle) {
      if (widgetConfig.title) {
        title = <h2 className={titleClasses}>{widgetConfig.title}</h2>;
      }
      else if (widgetConfig.displayTitle) {
        title = <h2 className={titleClasses}>{widgetConfig.displayTitle}</h2>;
      }
    }

    return title;
  }

  getStyles(widgetConfig = {}) {
    const styles = {};

    if (widgetConfig.backgroundColor) {
      styles.backgroundColor = widgetConfig.backgroundColor;
    }

    if (widgetConfig.backgroundImageUrl) {
      styles.backgroundImage = `url(${widgetConfig.backgroundImageUrl})`;
    }

    return styles;
  }

  getComponent(widgetConfig, widgetName) {
    const CurrentWidget = widgetComponents[widgetName];

    if (!CurrentWidget) {
      return (<span className="widget--not--found"/>);
    }

    // And we get the relevant state for that widget.
    const widgetReducerProp =
      this.props.widgetState[widgetName] ||
      this.props.widgetState[widgetName.replace('Widget', '')];

    return (
      <CurrentWidget
        widgetLocationName={this.props.widgetLocationName}
        widgetState={this.props.widgetState}
        widgetActions={this.props.widgetActions}
        widgetReducerProp={widgetReducerProp}
        widgetConfig={widgetConfig}/>
    );
  }

  render() {
    // First, find all widgets we want to render for this widgetcontainer
    let currentWidgetStates = this.props.widgetState.widgetLocations[this.props.widgetLocationName];
    if (currentWidgetStates) {
      // Loop over the widgets, you can have as many as you'd like
      const widgets = (Array.isArray(currentWidgetStates) ? currentWidgetStates : [currentWidgetStates]).map((currentWidgetState, idx) => {
        // Get widgetConfig
        const widgetConfig = currentWidgetState.widgetConfig;

        // Now we get the widget we wish to render.
        const widgetName = currentWidgetState.widgetName;

        // Get the component
        const CurrentWidget = this.getComponent(widgetConfig, widgetName);

        // Get a generic title
        const title = this.getTitle(widgetConfig, widgetName);

        // Get generic background image/style
        const widgetStyles = this.getStyles(widgetConfig);

        // We now render our widget inside a container
        return (
          <div
            className={`generic-widget-container ${widgetName}--container`}
            key={`${this.props.widgetLocationName}_${widgetName}_${idx}`}
          >
            {title}
            <div
              className={`${widgetName}--wrapper`}
              style={widgetStyles}
            >
              {CurrentWidget}
            </div>
          </div>
        );
      });

      // Finally we render the array of widgets relevant to this position.
      return (
        <div className={`${this.props.widgetLocationName}--generic-widget-container`}>
          {widgets}
        </div>
      );
    }

    // This renders if no widgets were found for this position
    return <span className={`${this.props.widgetLocationName}--generic-widget-container no-widgets`}/>;
  }
}

WidgetContainer.displayName = 'WidgetContainer';
WidgetContainer.propTypes = {
  widgetLocationName: PropTypes.string.isRequired,
  widgetState: PropTypes.object.isRequired,
  widgetActions: PropTypes.object.isRequired
};

export default WidgetContainer;
