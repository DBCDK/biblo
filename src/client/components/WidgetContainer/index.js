/**
 * @file
 * entry point for the preview page
 */

import {renderComponent} from '../../App';
import PreviewPage from './PreviewPage.component';

renderComponent(PreviewPage, 'content');
