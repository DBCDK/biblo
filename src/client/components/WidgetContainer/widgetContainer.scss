@import 'tools';

.generic-widget-container {
  margin-bottom: 3em;
}

.widget--title--generic {
  margin: 0;
  margin-bottom: 0.5em;
  margin-left: 1em;
  @include max-screen($tablet) {
    font-size: 1em;
  }
}

.generic-widget-wrapper {
  padding: 1em;
  position: relative;
}

.widget--campaign-logo {
  position: absolute;
  bottom: 100%;
  right: 0;
  margin-right: 1em;
  margin-bottom: 1em;

  img {
    display: block;
    max-height: 40px;
  }
}

.widget-element--profileimage {
  @extend %widget-profileimage;
}

.widget-element--main {
  position: relative;
}

.widget-element--author {
  white-space: nowrap;
  overflow: hidden;
  margin-bottom: 1em;
  font-size: 0.8em;
  line-height: 0.8em;
  font-weight: bold;

  .twemoji {
    width: 1em;
  }
}

.text--post {
  .widget-element--content {
    @include line-clamp(6);
    @extend %quote;
  }

  .widget-element--profileimage {
    position: relative;
    float: left;
    margin-bottom: -10px;
    display: inline-block;
  }
}

.video--post {
  .widget-element--content {
    @extend %video-overlay;
  }
}
