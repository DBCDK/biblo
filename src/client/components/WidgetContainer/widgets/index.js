/**
 * @file: A collection of widgets, collected for ease of use in WidgetContainer
 */

import {LatestReviewsWidget} from './LatestReviews/LatestReviews.widget.component';
import {DummyWidget} from './DummyWidget/Dummy.widget.component';
import {ColoredHeaderWidget} from './ColoredHeaderWidget/ColoredHeader.widget.component';
import {ContentGridWidget} from './ContentGridWidget/ContentGrid.widget.component';
import {ContentPageTextWidget} from './ContentPageTextWidget/ContentPageText.widget.component';
import {ContentPageImageWidget} from './ContentPageImageWidget/ContentPageImage.widget.component';
import {ContentPageEmbeddedVideoWidget} from './ContentPageEmbeddedVideoWidget/ContentPageEmbeddedVideo.widget.component';
import {FullWidthBannerWidget} from './FullWidthBannerWidget/FullWidthBanner.widget.component';
import {BestRatedWorksWidget} from './BestRatedWorksWidget/BestRatedWorks.widget.component';
import {LatestGroupPostsWidget} from './LatestGroupPostsWidget/LatestGroupPosts.widget.component';
import {FactBoxWidget} from './FactBoxWidget/FactBox.widget.component';
import {PopularGroupsWidget} from './PopularGroupsWidget/PopularGroups.widget.component.js';
import {EditoriallySelectedReviewsWidget} from './EditoriallySelectedReviewsWidget/EditoriallySelectedReviews.widget.component';
import {EditoriallySelectedMaterialsWidget} from './EditoriallySelectedMaterialsWidget/EditoriallySelectedMaterials.widget.component';

const widgetComponents = {
  FactBoxWidget,
  BestRatedWorksWidget,
  LatestReviewsWidget,
  DummyWidget,
  FullWidthBannerWidget,
  ColoredHeaderWidget,
  ContentGridWidget,
  ContentPageTextWidget,
  ContentPageImageWidget,
  ContentPageEmbeddedVideoWidget,
  LatestGroupPostsWidget,
  PopularGroupsWidget,
  EditoriallySelectedReviewsWidget,
  EditoriallySelectedMaterialsWidget
};

export default widgetComponents;
