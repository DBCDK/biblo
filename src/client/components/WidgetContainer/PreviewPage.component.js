/* eslint-disable react/no-danger */

import React from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import PageLayout from '../Layout/PageLayout.component.js';
import WidgetContainer from '../WidgetContainer/WidgetContainer.component';

import * as searchActions from '../../Actions/search.actions';
import * as widgetActions from '../../Actions/widget.actions';

export class PreviewPage extends React.Component {
  render() {
    return (
      <PageLayout searchState={this.props.searchState} searchActions={this.props.searchActions} profileState={this.props.profileState} globalState={this.props.globalState} >
      <WidgetContainer
          widgetLocationName="previewPage"
          widgetState={this.props.widgetState}
          widgetActions={this.props.widgetActions} />
      </PageLayout>
    );
  }
}

PreviewPage.displayName = 'PreviewPage';
PreviewPage.propTypes = {
  profileState: React.PropTypes.object.isRequired,
  widgetState: React.PropTypes.object.isRequired,
  widgetActions: React.PropTypes.object.isRequired,
  searchState: React.PropTypes.object.isRequired,
  searchActions: React.PropTypes.object.isRequired,
  globalState: React.PropTypes.object.isRequired
};

export default connect(
  // Map redux state to group prop
  (state) => {
    return {
      profileState: state.profileReducer,
      searchState: state.searchReducer,
      widgetState: state.widgetReducer,
      globalState: state.globalReducer
    };
  },

  // Map group actions to actions props
  (dispatch) => {
    return {
      searchActions: bindActionCreators(searchActions, dispatch),
      widgetActions: bindActionCreators(widgetActions, dispatch)
    };
  }
)(PreviewPage);
