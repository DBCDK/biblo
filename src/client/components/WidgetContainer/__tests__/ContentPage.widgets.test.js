/**
 * @file: This file contains tests for the widgets associated with ContentPages.
 **/

// import libs
import React from 'react';
import expect from 'expect';
import $ from 'teaspoon';

// import components
import WidgetContainer from '../WidgetContainer.component';

describe('Test ContentPages Widgets', () => {
  it('ContentPageText should render', () => {
    const widgetLocationName = 'test-widget-location--content-page-text';
    let widgetState = {
      widgetLocations: {}
    };
    const widgetActions = {};

    widgetState.widgetLocations[widgetLocationName] = {
      widgetName: 'ContentPageTextWidget',
      widgetConfig: {
        content: '<h2>Bob er sej!</h2>'
      }
    };

    let component = (
      <WidgetContainer
        widgetLocationName={widgetLocationName}
        widgetState={widgetState}
        widgetActions={widgetActions} />
    );

    let $root = $(component).render();

    const textInnerHTML = $root.find('.content-page--text-widget').props().dangerouslySetInnerHTML.__html;
    expect(textInnerHTML).toEqual('<h2>Bob er sej!</h2>');
  });

  it('ContentPageImage should render', () => {
    const widgetLocationName = 'content-page--image--test-widget-location';
    let widgetState = {
      widgetLocations: {}
    };
    const widgetActions = {};

    widgetState.widgetLocations[widgetLocationName] = {
      widgetName: 'ContentPageImageWidget',
      widgetConfig: {
        src: 'bob',
        alt: 'bob2',
        title: 'bob3'
      }
    };

    let component = (
      <WidgetContainer
        widgetLocationName={widgetLocationName}
        widgetState={widgetState}
        widgetActions={widgetActions} />
    );

    let $root = $(component).render();

    const renderedImage = $root.find('.content-page--image-widget > img').props();
    expect(renderedImage).toEqual({alt: 'bob2', src: 'bob', title: 'bob3'});
  });

  it('ContentPageEmbeddedVideo should render', () => {
    const widgetLocationName = 'content-page--video--test-widget-location';
    let widgetState = {
      widgetLocations: {}
    };
    const widgetActions = {};

    widgetState.widgetLocations[widgetLocationName] = {
      widgetName: 'ContentPageEmbeddedVideoWidget',
      widgetConfig: {type: 'YouTube', src: 'https://www.youtube.com/embed/qZ3fiOctBkE'}
    };

    let component = (
      <WidgetContainer
        widgetLocationName={widgetLocationName}
        widgetState={widgetState}
        widgetActions={widgetActions} />
    );

    let $root = $(component).shallowRender();

    const EmbeddedVideoDisplayName = $root.find('.generic-widget-container')['0'].props.children[1].props.children.type.displayName;
    expect(EmbeddedVideoDisplayName).toEqual('ContentPageEmbeddedVideoWidget');
  });
});
