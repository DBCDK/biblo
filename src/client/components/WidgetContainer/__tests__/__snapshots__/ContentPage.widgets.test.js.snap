// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ContentPages Widgets ContentPageEmbeddedVideo should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WidgetContainer
    widgetActions={Object {}}
    widgetLocationName="content-page--video--test-widget-location"
    widgetState={
      Object {
        "widgetLocations": Object {
          "content-page--video--test-widget-location": Object {
            "widgetConfig": Object {
              "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
              "type": "YouTube",
            },
            "widgetName": "ContentPageEmbeddedVideoWidget",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="generic-widget-container ContentPageEmbeddedVideoWidget--container"
        >
          <span
            className="widget--no--title"
          />
          <div
            className="ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper"
            style={Object {}}
          >
            <ContentPageEmbeddedVideoWidget
              widgetActions={Object {}}
              widgetConfig={
                Object {
                  "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                  "type": "YouTube",
                }
              }
              widgetIndex={0}
              widgetLocationName="content-page--video--test-widget-location"
              widgetReducerProp={undefined}
              widgetState={
                Object {
                  "widgetLocations": Object {
                    "content-page--video--test-widget-location": Object {
                      "widgetConfig": Object {
                        "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                        "type": "YouTube",
                      },
                      "widgetName": "ContentPageEmbeddedVideoWidget",
                    },
                  },
                }
              }
            />
          </div>
        </div>,
      ],
      "className": "content-page--video--test-widget-location--generic-widget-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "content-page--video--test-widget-location_ContentPageEmbeddedVideoWidget_0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="widget--no--title"
            />,
            <div
              className="ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper"
              style={Object {}}
            >
              <ContentPageEmbeddedVideoWidget
                widgetActions={Object {}}
                widgetConfig={
                  Object {
                    "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                    "type": "YouTube",
                  }
                }
                widgetIndex={0}
                widgetLocationName="content-page--video--test-widget-location"
                widgetReducerProp={undefined}
                widgetState={
                  Object {
                    "widgetLocations": Object {
                      "content-page--video--test-widget-location": Object {
                        "widgetConfig": Object {
                          "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                          "type": "YouTube",
                        },
                        "widgetName": "ContentPageEmbeddedVideoWidget",
                      },
                    },
                  }
                }
              />
            </div>,
          ],
          "className": "generic-widget-container ContentPageEmbeddedVideoWidget--container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "widget--no--title",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ContentPageEmbeddedVideoWidget
                widgetActions={Object {}}
                widgetConfig={
                  Object {
                    "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                    "type": "YouTube",
                  }
                }
                widgetIndex={0}
                widgetLocationName="content-page--video--test-widget-location"
                widgetReducerProp={undefined}
                widgetState={
                  Object {
                    "widgetLocations": Object {
                      "content-page--video--test-widget-location": Object {
                        "widgetConfig": Object {
                          "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                          "type": "YouTube",
                        },
                        "widgetName": "ContentPageEmbeddedVideoWidget",
                      },
                    },
                  }
                }
              />,
              "className": "ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper",
              "style": Object {},
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "widgetActions": Object {},
                "widgetConfig": Object {
                  "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                  "type": "YouTube",
                },
                "widgetIndex": 0,
                "widgetLocationName": "content-page--video--test-widget-location",
                "widgetReducerProp": undefined,
                "widgetState": Object {
                  "widgetLocations": Object {
                    "content-page--video--test-widget-location": Object {
                      "widgetConfig": Object {
                        "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                        "type": "YouTube",
                      },
                      "widgetName": "ContentPageEmbeddedVideoWidget",
                    },
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="generic-widget-container ContentPageEmbeddedVideoWidget--container"
          >
            <span
              className="widget--no--title"
            />
            <div
              className="ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper"
              style={Object {}}
            >
              <ContentPageEmbeddedVideoWidget
                widgetActions={Object {}}
                widgetConfig={
                  Object {
                    "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                    "type": "YouTube",
                  }
                }
                widgetIndex={0}
                widgetLocationName="content-page--video--test-widget-location"
                widgetReducerProp={undefined}
                widgetState={
                  Object {
                    "widgetLocations": Object {
                      "content-page--video--test-widget-location": Object {
                        "widgetConfig": Object {
                          "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                          "type": "YouTube",
                        },
                        "widgetName": "ContentPageEmbeddedVideoWidget",
                      },
                    },
                  }
                }
              />
            </div>
          </div>,
        ],
        "className": "content-page--video--test-widget-location--generic-widget-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "content-page--video--test-widget-location_ContentPageEmbeddedVideoWidget_0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="widget--no--title"
              />,
              <div
                className="ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper"
                style={Object {}}
              >
                <ContentPageEmbeddedVideoWidget
                  widgetActions={Object {}}
                  widgetConfig={
                    Object {
                      "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                      "type": "YouTube",
                    }
                  }
                  widgetIndex={0}
                  widgetLocationName="content-page--video--test-widget-location"
                  widgetReducerProp={undefined}
                  widgetState={
                    Object {
                      "widgetLocations": Object {
                        "content-page--video--test-widget-location": Object {
                          "widgetConfig": Object {
                            "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                            "type": "YouTube",
                          },
                          "widgetName": "ContentPageEmbeddedVideoWidget",
                        },
                      },
                    }
                  }
                />
              </div>,
            ],
            "className": "generic-widget-container ContentPageEmbeddedVideoWidget--container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "widget--no--title",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ContentPageEmbeddedVideoWidget
                  widgetActions={Object {}}
                  widgetConfig={
                    Object {
                      "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                      "type": "YouTube",
                    }
                  }
                  widgetIndex={0}
                  widgetLocationName="content-page--video--test-widget-location"
                  widgetReducerProp={undefined}
                  widgetState={
                    Object {
                      "widgetLocations": Object {
                        "content-page--video--test-widget-location": Object {
                          "widgetConfig": Object {
                            "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                            "type": "YouTube",
                          },
                          "widgetName": "ContentPageEmbeddedVideoWidget",
                        },
                      },
                    }
                  }
                />,
                "className": "ContentPageEmbeddedVideoWidget--wrapper generic-widget-wrapper",
                "style": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "widgetActions": Object {},
                  "widgetConfig": Object {
                    "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                    "type": "YouTube",
                  },
                  "widgetIndex": 0,
                  "widgetLocationName": "content-page--video--test-widget-location",
                  "widgetReducerProp": undefined,
                  "widgetState": Object {
                    "widgetLocations": Object {
                      "content-page--video--test-widget-location": Object {
                        "widgetConfig": Object {
                          "src": "https://www.youtube.com/embed/qZ3fiOctBkE",
                          "type": "YouTube",
                        },
                        "widgetName": "ContentPageEmbeddedVideoWidget",
                      },
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
